src/hiccup/compiler.clj

```diff
@@ -13,17 +14,33 @@
 (defn- end-tag []
   (if (xml-mode?) " />" ">"))
 
+(defn- render-attr-value [value]
+  (cond
+    (string? value)
+    (util/escape-html (util/as-str value))
+    (map? value)
+    (str
+      (->> value
+           (mapv (fn [[k v]] (str (util/as-str k) ":" (util/as-str v))))
+           sort
+           (string/join ";"))
+      ";")
+    (sequential? value)
+    (string/join " " value)
+    :default
+    (util/escape-html (util/as-str value))))
+
 (defn- xml-attribute [name value]
-  (str " " (util/as-str name) "=\"" (util/escape-html value) "\""))
+  (str " " (util/as-str name) "=\"" (render-attr-value value) "\""))
 
 (defn- render-attribute [[name value]]
   (cond
     (true? value)
-      (if (xml-mode?)
```

Comment:
Formatting changes must be removed from the PR. The PR should cover only the changes necessary to implement the feature.